packages/amplify-graphql-api-construct-tests/src/sql-tests-common/sql-models-auto-increment.ts

```diff
@@ -0,0 +1,187 @@
+
+      expect(getUpdatedCoffeeOrder1.orderNumber).toEqual(createCoffeeOrder1.orderNumber);
+      expect(getUpdatedCoffeeOrder1.order).toEqual('hot brew');
+
+      // List Todo Query
+      const createCofffeeOrder2 = await coffeeQueueTableCRUDLHelper.create({ order: 'latte' });
+      const listTodo = await coffeeQueueTableCRUDLHelper.list();
+
+      expect(listTodo.items.length).toEqual(2);
+      expect(listTodo.items).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            orderNumber: getUpdatedCoffeeOrder1.orderNumber,
+            order: 'hot brew',
+          }),
+          expect.objectContaining({
+            orderNumber: createCofffeeOrder2.orderNumber,
+            order: 'latte',
+          }),
+        ]),
+      );
+
+      // Delete Todo Mutation
+      const deleteCoffeeOrder1 = await coffeeQueueTableCRUDLHelper.delete({ orderNumber: createCoffeeOrder1.orderNumber });
+
+      expect(deleteCoffeeOrder1.orderNumber).toEqual(createCoffeeOrder1.orderNumber);
+      expect(deleteCoffeeOrder1.order).toEqual('hot brew');
+
+      const getDeletedCoffeeOrder1 = await coffeeQueueTableCRUDLHelper.get({ orderNumber: createCoffeeOrder1.orderNumber });
+
+      expect(getDeletedCoffeeOrder1).toBeNull();
+
+      // List Todo Query after delete
+      const listCoffeeOrdersAfterDelete = await coffeeQueueTableCRUDLHelper.list();
+
+      expect(listCoffeeOrdersAfterDelete.items.length).toEqual(1);
+      expect(listCoffeeOrdersAfterDelete.items).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            orderNumber: createCofffeeOrder2.orderNumber,
+            order: 'latte',
+          }),
+        ]),
+      );
+
+      // Check invalid CRUD operation returns generic error message
+      const createTodo6 = await coffeeQueueTableCRUDLHelper.create({ order: 'mocha' });
+
+      try {
+        await coffeeQueueTableCRUDLHelper.create({ orderNumber: createTodo6.orderNumber, order: 'americano' });
+      } catch (error) {
+        console.log('throw in create');
+        coffeeQueueTableCRUDLHelper.checkGenericError(error?.message);
+      }
+
+      const invalidOrderNumber = 99999999;
+
+      try {
+        await coffeeQueueTableCRUDLHelper.get({ orderNumber: invalidOrderNumber });
+      } catch (error) {
+        console.log('throw in get');
+        coffeeQueueTableCRUDLHelper.checkGenericError(error?.message);
+      }
+
+      try {
+        await coffeeQueueTableCRUDLHelper.update({ orderNumber: invalidOrderNumber, order: 'cortado' });
+      } catch (error) {
+        console.log('throw in update');
+        coffeeQueueTableCRUDLHelper.checkGenericError(error?.message);
+      }
+
+      try {
+        await coffeeQueueTableCRUDLHelper.delete({ orderNumber: invalidOrderNumber });
+      } catch (error) {
+        console.log('throw in delete');
+        coffeeQueueTableCRUDLHelper.checkGenericError(error?.message);
+      }
+    });
```

Comment:
Does PG allow someone to specify a value for a `SERIAL` field? If so, we need a test for that case. If not, we need a test that a customer cannot supply a value for it (which might just be a unit test that asserts the serial default field is not present in the input shape.

Reply:
PG does allow users to specify a value for `SERIAL` fields (but does not stop users' sequences from producing that value later). I modified the test to use a custom value for one of the inserts.